### PR TITLE
[doc] add product diagrams

### LINKED ISSUE #1347: [doc]: Publish product diagrams
**Describe the request**

Product diagrams for PWA-Studio only exist in issue comments and in the internal wiki.
They need to be available in the docs site.

**Possible solutions**

Publish infographics from the Product Diagrams internal wiki page.

* Need to find or create appropriate page for these diagrams
* Finalize styles used in the diagrams
* Optimize image before uploading

**Screenshots**

n/a

**Please let us know whether this is a new topic or a topic change request:**
 - [x] New Topic Request (ie. missing entire topic/section)
 - [ ] Topic Change Request (ie. spelling, organization)

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

* Add the product diagrams to appropriate topics in the doc site.
* Update topic content to reflect current state of the project.
* Clean up any linting errors in the modified topics.

## Related Issue

Closes #1347 

## How have YOU tested this?

See verification steps

## Verification Steps

1. Navigate to the `pwa-devdocs` directory
2. Run linter on the technology overview file:
    ```sh
   npm run lint src/technologies/overview/index.md
   ```
3. Verify no linting issues associated with this file.
4. Run linter on the tools and library file:
    ```sh
    npm run lint src/technologies/tools-libraries/index.md
    ```
5. Verify no linting issues associated with this file.
6. Build and run the preview server: `npm run develop`
7. Navigate to `/technologies/overview/`
8. Verify the developer overview image is on the page.
9. Navigate to `/technologies/tools-libraries/`
10. Verify the tools and technology image is on the page.

## Checklist:

- [x] I have updated the documentation accordingly, if necessary.
- [ ] I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/technologies/tools-libraries/index.md [hunk 1]
```diff
@@ -2,15 +2,41 @@
...
-| ------------------- | :-------------------------------------------: | :-------------------------------------------: | :-------------------------------------------: |
-| [webpack](#webpack) | <i class="material-icons green">check_box</i> |                                               |                                               |
-| [React](#react)     | <i class="material-icons green">check_box</i> | <i class="material-icons green">check_box</i> |                                               |
-| [Redux](#redux)     | <i class="material-icons green">check_box</i> | <i class="material-icons green">check_box</i> |                                               |
-| [GraphQL](#graphql) | <i class="material-icons green">check_box</i> | <i class="material-icons green">check_box</i> | <i class="material-icons green">check_box</i> |
-{:style="table-layout:auto"}
+|                           |                  Contributor                  |              Frontend developer               |         Sysadmin/Devops/Backend developer          |
```
Comment (supernova-at) [inline pwa-devdocs/src/technologies/tools-libraries/index.md]: What's the difference between a contributor and a front end developer?

Comment (jcalcaben) [reply pwa-devdocs/src/technologies/tools-libraries/index.md]: I see a contributor as also having backend scripting skills to work on packages such buildpack or upward-js

**Comment by supernova-at** [comment]
I was able to verify steps 1-5 but not past that (I may need Calcaben's help setting up my docs installation, I'm pretty sure it's something on my side).

As such, can whoever is QAing these please also run the verification steps?

**Comment by dpatil-magento** [comment]
@supernova-at I have that setup with bundler gem. So, you can skip these tickets to QA always. If you need bundler setup I can look into that as well.

**Pull request merged**